gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/config/KubernetesConfig.java

```diff
@@ -328,7 +323,35 @@ private String getHomeDir() {
         return System.getProperty("user.home", ".");
     }
 
-    private String getKubeConfigFilename() {
+    private static String getWinHomeDir() {
+        String homeDrive = System.getenv("HOMEDRIVE");
+        String homePath = System.getenv("HOMEPATH");
+        if (homeDrive != null && !homeDrive.isEmpty() && homePath != null && !homePath.isEmpty()) {
+            String homeDir = homeDrive + homePath;
+            File f = new File(homeDir);
+            if (f.exists() && f.isDirectory()) {
+                return homeDir;
+            }
+        }
+        String userProfile = System.getenv("USERPROFILE");
+        if (userProfile != null && !userProfile.isEmpty()) {
+            File f = new File(userProfile);
+            if (f.exists() && f.isDirectory()) {
+                return userProfile;
+            }
+        }
+        return null;
+    }
+
+    private String getKubeConfigFilename(String overrideFile) {
+        // if an override file is given then, it becomes mandatory
+        if (overrideFile != null) {
+            if (new File(overrideFile).isFile()) {
+                return overrideFile;
+            }
+            throw new IllegalArgumentException(String.format("Override Kubernetes config file '%s' does not exists", overrideFile));
+        }
```

Comment:
No it is not, my take was to keep the existing as is and throw an exception if an explicit file is set. If you set anything, this detects its in the cluster or not, and if not that it can work. The default client always exists, because it is a singleton (see `io.gravitee.kubernetes.client.spring.KubernetesClientConfiguration`) so it should not fail else the Gateway might not start. The new constructor might pass a file, in that case, it is intentional, thus it must  work imo.

Reply:
But one might agree that it should fail always if none of the three ways work when not using the default constructor. Might add something like that

Reply:
WDYT ?

Reply:
OK I understand 👍